codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/SerializeMiddleware.java

```diff
@@ -0,0 +1,165 @@
+
+                func (op *$opName:L) ID() string {
+                    return "OperationSerializer"
+                }
+
+                $handleSerialize:W
+
+                """,
+                MapUtils.of(
+                    "opName", serialName,
+                    "handleSerialize", generateHandleSerialize()
+        ));
+    }
+
+    private GoWriter.Writable generateHandleSerialize() {
+        return goTemplate(
+                """
+
+                        func (op *$opName:L) HandleSerialize (ctx $context:T, in $input:T, next $handler:T) (
+                        out $output:T, metadata $metadata:T, err error) {
+
+                            $body:W
+
+                            return next.HandleSerialize(ctx, in)
+                        }
+
+                        """, MapUtils.of(
+                        "context", SmithyGoDependency.CONTEXT.interfaceSymbol("Context"),
+                        "input", SmithyGoDependency.SMITHY_MIDDLEWARE.struct("SerializeInput"),
+                        "handler", SmithyGoDependency.SMITHY_MIDDLEWARE.struct("SerializeHandler"),
+                        "output", SmithyGoDependency.SMITHY_MIDDLEWARE.struct("SerializeOutput"),
+                        "metadata", SmithyGoDependency.SMITHY_MIDDLEWARE.struct("Metadata"),
+                        "opName", serialName,
+                        "body", generateHandleSerializeBody())
+        );
+    }
+
+    private GoWriter.Writable generateHandleSerializeBody() {
+        return goTemplate("""
+                    $route:W
+
+                    $serialize:W
+
+                """,
+                MapUtils.of(
+                        "route", handleHttp(),
+                        "serialize", handlePayload()
+                ));
+    }
+
+    private GoWriter.Writable handleHttp() {
+        String contentType = "application/x-amz-json-1.0";
+
+        return goTemplate("""
+                    input, ok := in.Parameters.($input:P)
+                    if !ok {
+                        return out, metadata, $errorf:T("unexpected input type %T", in.Parameters)
+                    }
+                    _ = input
+
+                    req, ok := in.Request.($request:P)
+                    if !ok {
+                        return out, metadata, $errorf:T("unexpected transport type %T", in.Request)
+                    }
+
+                    req.Method = $methodPost:T
+                    req.URL.Path = "/"
+                    req.Header.Set("Content-Type", $contentType:S)
+                    req.Header.Set("X-Amz-Target", $target:S)
+
+                """,
+        MapUtils.of(
+                "input", ctx.getSymbolProvider().toSymbol(input),
+                "request", SMITHY_HTTP_TRANSPORT.pointableSymbol("Request"),
+                "methodPost", GoStdlibTypes.Net.Http.MethodPost,
+                "contentType", contentType,
+                "target", ctx.getService().getId().getName() + operation.getId().getName(),
```

Comment:
You may be missing a dot

> The value of this header is the [shape name](https://smithy.io/2.0/spec/model.html#grammar-token-smithy-Identifier) of the service's [Shape ID](https://smithy.io/2.0/spec/model.html#shape-id) joined to the [shape name](https://smithy.io/2.0/spec/model.html#grammar-token-smithy-Identifier) of the operation's [Shape ID](https://smithy.io/2.0/spec/model.html#shape-id), separated by a single period (.) character.
For example, the value for the operation ns.example#MyOp of the service ns.example#MyService is MyService.MyOp.

https://smithy.io/2.0/aws/protocols/aws-json-1_0-protocol.html